### DISCUSSION TITLE
i2cdev one dev per I2C port or one dev per I2C device

### DISCUSSION THREAD

**Discussion by mbastida123** [UncleRus/esp-idf-lib#426] [Q&A] [answered]
Hi,

We are using the I2Cdev library to connect to two devices on the same I2C bus.

The question is, each I2C has to have a `i2c_dev_t` struct or both devices have to share the same `i2c_dev_t`.

Inspecting how the library is built I would say it's the first one. But then I get to the conclusión that esp-idf's I2C library is not thread-safe when accessing the same device from diferent threads. That's why in i2cdev the mutex is not shared accross diferent I2C devices.

Am I right?

**Comment by UncleRus** [comment, marked as answer]
Each I2C device must have its own descriptor.

**Reply by mbastida123** [reply]
Hi @UncleRus,

Thanks for the quick response!
